components/Search/results/ResidentsTable.spec.tsx

```diff
@@ -1,14 +1,47 @@
 import { render } from '@testing-library/react';
 
-import ResidenTable from './ResidentsTable';
-import { mockedLegacyResident } from 'factories/residents';
-
-describe('ResidenTable component', () => {
-  const props = {
-    records: [mockedLegacyResident],
-  };
-  it('should render properly', () => {
-    const { asFragment } = render(<ResidenTable {...props} />);
+import ResidentsTable from './ResidentsTable';
+import {
+  legacyResidentFactory,
+  mockedLegacyResident,
+} from 'factories/residents';
+import { UserContext } from '../../UserContext/UserContext';
+import { mockedUser, userFactory } from '../../../factories/users';
+
+describe('ResidentsTable component', () => {
+  it('should render a list of residents', () => {
+    const { asFragment } = render(
+      <UserContext.Provider value={{ user: mockedUser }}>
+        <ResidentsTable records={[mockedLegacyResident]} />
+      </UserContext.Provider>
+    );
     expect(asFragment()).toMatchSnapshot();
   });
+
+  it('should render a restricted flag against any residents that the current user cannot manage cases for', () => {
+    const { getByText } = render(
+      <UserContext.Provider
+        value={{
+          user: userFactory.build({
+            hasAdminPermissions: false,
+            hasDevPermissions: false,
+            hasChildrenPermissions: false,
+            hasUnrestrictedPermissions: false,
+            hasAllocationsPermissions: false,
+            hasAdultPermissions: true,
+          }),
+        }}
+      >
+        <ResidentsTable
+          records={[
+            legacyResidentFactory.build({
+              ageContext: 'C',
+            }),
+          ]}
+        />
+      </UserContext.Provider>
+    );
+
+    getByText('RESTRICTED');
```

Comment:
Could I check my understanding? I'm not sure I understand how the assertion works in this test.

Are we defining a `getByText` component that sets up a user without child permissions and creates a list of residents under the children group? Why do we pass in `RESTRICTED` as text to `getByText`?

Reply:
`getByText` is a method we get once we render our component, provided by React Testing Library – https://testing-library.com/docs/queries/bytext

What we're saying here is:
* Render the component
* Look for an element with the text "RESTRICTED" in it

`getByText` with throw an error if it doesn't find anything, which fails the test, so we don't explicitly need an `expect()` statement, but if we did it would look like:

```ts
expect(getByText("RESTRICTED").toBeTruthy();
```

Reply:
I see, thanks! 🙌🏾